### PR TITLE
Internal: Allow Filtering Common Controls [ED-16100]

### PR DESCRIPTION
**Pull request by nuritsha**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

**Review by ramiy — COMMENTED** [review]
not sure, but if you can change the value, shouldn't we use 'let'.

**Comment by nuritsha** [comment]
> not sure, but if you can change the value, shouldn't we use 'let'.

@ramiy `const` should work here, the value can be changed by the filter, but the `commonControls` variable is initialized only after the filter returns. I don't think we want to allow modifying this value after the filter.

File: assets/dev/js/editor/editor-base.js [hunk 1]
```diff
@@ -287,7 +287,8 @@ export default class EditorBase extends Marionette.Application {
 			}
 
 			if ( ! this.widgetsCache[ widgetType ].commonMerged && ! this.widgetsCache[ widgetType ].atomic_controls ) {
-				jQuery.extend( this.widgetsCache[ widgetType ].controls, this.widgetsCache.common.controls );
+				const commonControls = elementor.hooks.applyFilters( 'elements/widget/controls/common/default', this.widgetsCache.common.controls, widgetType );
```
Comment (rami-elementor) [inline assets/dev/js/editor/editor-base.js:290]:
```suggestion
				let commonControls = this.widgetsCache.common.controls;

				/**
				 * Filter widgets common controls.
				 *
				 * @param array  commonControls - An array of the default common controls.
				 * @param string widgetType     - The widget type.
				 */
				commonControls = elementor.hooks.applyFilters( 'elements/widget/controls/common/default', commonControls, widgetType );
```

Comment (louiswol94) [inline assets/dev/js/editor/editor-base.js:290]: @nuritsha , I would be interested to see the use case for this :) But anyway, I think this filter might be too early. In the "Optimized Markup" experiment, we introduced a new Advanced tab `common-optimized`, so you might need to add the filter after that. Maybe just double check?

Comment (nuritsha) [reply assets/dev/js/editor/editor-base.js]: @louiswol94 I don't see any reference to that experiment in the js code, can you please direct me to the relevant place?

Comment (louiswol94) [reply assets/dev/js/editor/editor-base.js]: @nuritsha The experiment name is `e_optimized_markup`. You will see on [line 302](https://github.com/elementor/elementor/blob/e321132bc41340a2a4af4012aabaa8ec85e6afdc/assets/dev/js/editor/editor-base.js#L302) we merge those controls in as well. So I am thinking the filter should only be added after that?

Comment (nuritsha) [reply assets/dev/js/editor/editor-base.js]: yep, u r right

**Review by louiswol94 — APPROVED** [review]

**Pull request merged**